src/plugins/data/public/ui/dataset_selector/dataset_selector.tsx

```diff
@@ -134,13 +134,13 @@ export const DatasetSelector = ({
       button={
         <EuiToolTip content={`${selectedDataset?.title ?? 'Select data'}`}>
           <EuiButtonEmpty
-            className="datasetSelector__button"
+            className="datasetSelector__button datasetSelector__button--truncate"
             iconType="arrowDown"
             iconSide="right"
             onClick={togglePopover}
           >
             <EuiIcon type={datasetIcon} className="datasetSelector__icon" />
-            {datasetTitle}
+            <span className="datasetSelector__title">{datasetTitle}</span>
```

Comment:
could this actually use an EuiText and pass a classname `className="eui-textTruncate"` from OUI?

Reply:
I just gave a try, I found that the EuiText works, however, the truncate class seems won't be applied.

Reply:
actually, I also found out that after applying the EuiText the Icon happens to be at the top of the title.
<img width="1211" alt="Screenshot 2024-09-20 at 1 13 50 PM" src="https://github.com/user-attachments/assets/7a9d35f7-a46d-421b-8b8a-11e737d97733">